### PR TITLE
fix: Don't attempt publishing non-latest published versions

### PR DESCRIPTION
**Pull request by Gudahtt**
When deciding whether to attempt publishing a package, this action would previously compare the latest published version with the current version. If they don't match, it would try to publish the package. However, that version may have been published already, just not as the latest version.

This is especially problematic for backports within monorepos. This would result in errors if any package was already published but not at the latest version.

The script has been updated to check whether the given package version was published at all. It no longer matters whether it was latest or not.

This also fixes the "update-packages-test", which tests with an old version of the snaps monorepo. It was broken because all of the packages being tested have had releases since the last successful run.

### CODE REVIEW

File: .github/workflows/build-lint-test.yml [hunk 1]
```diff
@@ -58,6 +58,7 @@ jobs:
             MANIFEST="${1}/package.json"
             MANIFEST_TEMP="${MANIFEST}_temp"
             VERSION=$(jq --raw-output .version "$MANIFEST")
+            echo "package: ${1}, operation: ${2}, current version: ${VERSION}"
```
Comment (Mrtenz) [inline .github/workflows/build-lint-test.yml:61]:
```suggestion
```

File: .github/workflows/build-lint-test.yml [hunk 2]
```diff
@@ -79,6 +80,7 @@ jobs:
               # e.g. 1.0.0 -> 1.0.1
               NEW_VERSION="${MAJOR}.${MINOR}.$((PATCH + 1))"
             fi
+            echo "Updated version: ${NEW_VERSION}"
```
Comment (Mrtenz) [inline .github/workflows/build-lint-test.yml:83]:
```suggestion
```

File: action.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ description: 'Publish the release'
 
 inputs:
   npm-tag:
-    description: 'The NPM tag to use when checking for the latest published version of a package. Defaults to `latest`.'
+    description: '(Deprecated) This option is ignored, and will be removed in a future release'
```
Comment (Mrtenz) [inline action.yml:6]: Why is this deprecated?

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: .github/workflows/build-lint-test.yml (hunk 2 above)
Comment (Gudahtt) [reply .github/workflows/build-lint-test.yml:83]: This log was useful for me understanding what the test is doing. See the test output:

```
package: packages/multichain-provider, operation: bump, current version: 0.32.2
Updated version: 0.32.3
package: packages/rpc-methods, operation: bump, current version: 0.32.2
Updated version: 0.32.3
package: packages/snaps-browserify-plugin, operation: unbump, current version: 0.32.2
Updated version: 0.32.1
package: packages/examples, operation: bump, current version: 0.32.2
Updated version: 0.32.3
```

The test is really quite difficult to understand without these logs. The test bumps packages at index 0-3, but it's not clear to the reader what those packages are (which is critical for understanding why the examples are skipped, for example).

File: action.yml (hunk 3 above)
Comment (Gudahtt) [reply action.yml:6]: As you can see in the diff for that script, we no longer reference the tag anywhere. The script no longer looks at any dist-tags, it doesn't need to.

File: .github/workflows/build-lint-test.yml (hunk 1 above)
Comment (Gudahtt) [reply .github/workflows/build-lint-test.yml:61]: See this comment: https://github.com/MetaMask/action-publish-release/pull/91#discussion_r1340219276

File: action.yml (hunk 3 above)
Comment (Mrtenz) [reply action.yml:6]: I mean, shouldn't the tags be checked here?

https://github.com/MetaMask/action-publish-release/pull/91/files#diff-5b483cdb3187bbb9b2f4fbd9ddec7bec678226279669903b0f6e7c1ba586e4c6R34-R35

Comment (Gudahtt) [reply action.yml:6]: For what purpose?

Comment (Mrtenz) [reply action.yml:6]: Ah, I think I understand how it works now. Sorry for the confusion!

Comment (Gudahtt) [reply action.yml:6]: No problem! Yeah it's a bit hard to follow, the purpose/basic functionality of the script is changing here but it's not documented anywhere.

Comment (Gudahtt) [reply action.yml:6]: I have added a description here to make that more clear: https://github.com/MetaMask/action-publish-release/pull/91/commits/4bdd232108a06edc05a0cc2e4aea41ef3b8f287e

File: scripts/get-release-packages.sh [hunk 4]
```diff
@@ -5,7 +5,13 @@ set -e
 set -u
 set -o pipefail
 
-tag="${1:-latest}"
+# ============================================================================
+# This script determines which packages to publish in a monorepo. It filters
+# the collection of package metadata passed in to include just the packages to
+# be published, then prints that filtered list.
+#
+# A package will be published if it cannot be found on the npm registry.
```
Comment (Gudahtt) [inline scripts/get-release-packages.sh:13]:
```suggestion
# A package will be published if it cannot be found on the npm registry at the
# current version.
```

File: action.yml (hunk 3 above)
Comment (Gudahtt) [inline action.yml:6]: This might be a bit more clear as well:
```suggestion
    description: '(Deprecated) This option is no longer used, and will be removed in a future release'
```

**Review by Mrtenz — APPROVED** [review]
